### ISSUE TITLE
sql: Executing SELECT * FROM information_schema.statistics group by index_name causes cluster to crash

### ISSUE THREAD

**Issue by roncrdb** [cockroachdb/cockroach#44689]
**Describe the problem**

On a local cluster, I ran select * from information_schema.statistics group by index_name; with no load and no data and it immediately crashed the cluster. A user has also reported this issue and they were able to reliably reproduce this behaviour as well.  Here is the stack trace: 

```
 select * from information_schema.statistics group by index_name;
*
* ERROR: [n1,client=127.0.0.1:51658,user=root] a SQL panic has occurred while executing "SELECT * FROM information_schema.statistics GROUP BY index_name": no indexes
*
*
* ERROR: [n1,client=127.0.0.1:51658,user=root] a panic has occurred!
*
panic while executing 1 statements: SELECT * FROM _._ GROUP BY _

goroutine 2638 [running]:
runtime/debug.Stack(0x8145ba0, 0xc0033ded40, 0x3)
	/usr/local/go/src/runtime/debug/stack.go:24 +0x9d
github.com/cockroachdb/cockroach/pkg/util/log.ReportPanic(0x8145ba0, 0xc0033ded40, 0xc000513500, 0x760e980, 0xc0009758f0, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/crash_reporting.go:217 +0xb6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc006829000, 0x8145ba0, 0xc0033ded40, 0x71f3b40, 0x8071330)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:724 +0x2a0
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc006829000, 0x8145ba0, 0xc0033ded40)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:442 +0x61
panic(0x71f3b40, 0x8071330)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).Build.func1(0xc00758e6c8)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:171 +0xae
panic(0x71f3b40, 0x8071330)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/pkg/sql.(*optVirtualTable).Index(0xc003f898b0, 0x0, 0x4c12d1f, 0xc006a22300)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt_catalog.go:1350 +0x39
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).allowImplicitGroupingColumn(0xc00117e300, 0xc000000001, 0xc003dcf780, 0x3)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/groupby.go:819 +0xc2
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildScalar(0xc00117e300, 0x817fc80, 0xc000a18000, 0xc003aeb8f8, 0xc003aeb9f0, 0xc000a19500, 0x0, 0x1, 0xc003aeb9f0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scalar.go:80 +0x33d7
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildProjectionList(0xc00117e300, 0xc003aeb8f8, 0xc003aeb9f0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/project.go:164 +0x68
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildSelectClause(0xc00117e300, 0xc003ac88f0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc003aeb800, 0x8145c60)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:835 +0x33a
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildSelect(0xc00117e300, 0xc0039fc180, 0x0, 0x0, 0x0, 0xc003aeb800, 0xc00117e300)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:759 +0x347
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmt(0xc00117e300, 0x815d320, 0xc0039fc180, 0x0, 0x0, 0x0, 0xc003aeb800, 0x400e818)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:241 +0x20d
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).Build(0xc00117e300, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:185 +0x10b
github.com/cockroachdb/cockroach/pkg/sql.(*optPlanningCtx).buildExecMemo(0xc006829b08, 0x8145c60, 0xc000975740, 0x0, 0x0, 0xc00758e848)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:419 +0x152
github.com/cockroachdb/cockroach/pkg/sql.(*planner).makeOptimizerPlan(0xc006829370, 0x8145c60, 0xc000975740, 0x1, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:161 +0x7d
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).makeExecPlan(0xc006829000, 0x8145c60, 0xc000975740, 0xc006829370, 0x10, 0x10)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:743 +0x10d
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc006829000, 0x8145c60, 0xc000975740, 0xc006829370, 0xd194c68, 0xc000535c68, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:625 +0xfa
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc006829000, 0x8145c60, 0xc000975740, 0x815d320, 0xc0039fc180, 0xc006ed20de, 0x3f, 0x0, 0x1, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:416 +0xc01
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc006829000, 0x8145c60, 0xc000975740, 0x815d320, 0xc0039fc180, 0xc006ed20de, 0x3f, 0x0, 0x1, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:98 +0x4ec
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc006829000, 0x8145ba0, 0xc003dcf740, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1243 +0x1aa2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc006829000, 0x8145ba0, 0xc0033ded40, 0xc00051aa00, 0x5400, 0x15000, 0xc00051aa98, 0xc00699efd0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1172 +0x1a3
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc00049b400, 0x8145ba0, 0xc0033ded40, 0xc006829000, 0x5400, 0x15000, 0xc00051aa98, 0xc00699efd0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:444 +0xce
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc000477539, 0xc00037fd00, 0x8145ba0, 0xc0033ded40, 0xc00699efd0, 0xc00049b400, 0xc000535600, 0x815b660, 0xc00037fce0, 0xc003e24360, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:584 +0x21c
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:513 +0x17b

*
* ERROR: [n1,client=127.0.0.1:51658,user=root] Reported as error c742bad30fc741ba9175c0d2e57e69d9
*
driver: bad connection
warning: connection lost!
opening new connection: all session settings will be lost
warning: error retrieving the transaction status: dial tcp [::1]:26257: connect: connection refused
warning: connection lost!
opening new connection: all session settings will be lost
warning: error retrieving the database name: dial tcp [::1]:26257: connect: connection refused
```

**Environment:**
 - CockroachDB 19.2.2

**Comment by awoods187** [comment]
This repros on master as well

**Comment by RoachietheSupportRoach** [comment]
Zendesk ticket #[4621](https://cockroachdb.zendesk.com/agent/tickets/4621) has been linked to this issue.

**Comment by solongordon** [comment]
At a glance this looks related to #41732, which is new in v19.2.2. I think this should be an invalid query because it's not grouping by the primary key. Assigning to @RaduBerinde for triage.

**Comment by RaduBerinde** [comment]
This came in on Sentry as #44659. A PR is out already. Closing this as a dup.

**Comment by roncrdb** [comment]
@RaduBerinde when will the PR be merged and released?

**Comment by RaduBerinde** [comment]
It will be merged very soon, so it will be in 19.2.4. Even with the fix, this query will return a `column "..." must appear in the GROUP BY clause or be used in an aggregate function` error anyway. You can't select all columns from the table when you're only grouping on a single one (the only exception is if that column is the PK, but virtual tables don't have PKs).

You can just do `SELECT DISTINCT ON (index_name) * FROM information_schema.statistics`. If `index_name` is not unique you will only get one row for that `index_name` (in this case the original query made no sense anyway).

**Comment by roncrdb** [comment]
Thank you @RaduBerinde !
